### PR TITLE
JAMES-3869 SPF mailet: exclude private networks in addition to 127.0.…

### PR DESCRIPTION
**Pull request by ouvtam**
…0.1 and make it configurable

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/SPF.java [hunk 1]
```diff
@@ -50,32 +58,52 @@
...
     public void init() {
+        debug = Boolean.parseBoolean(getInitParameter("debug", "false"));
         addHeader = Boolean.parseBoolean(getInitParameter("addHeader", "false"));
-        SPFLoggerAdapter logger = new SPFLoggerAdapter(Boolean.parseBoolean(getInitParameter("debug", "false")));
+        spf = new DefaultSPF(new SPFLoggerAdapter(debug));
+
+        Collection<String> ignoredNetworks = Splitter.on(',')
+            .trimResults()
+            .omitEmptyStrings()
+            .splitToList(getInitParameter("ignoreNetworks", privateNetworks));
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/SPF.java:102]: ignoredNetworks ?

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/SPFTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            public InetAddress getLocalHost() {
+                throw new UnsupportedOperationException("not supported");
+            }
+
+            @Override
+            public String getHostName(InetAddress addr) {
+                throw new UnsupportedOperationException("not supported");
+            }
+        };
+    }
```
Comment (chibenwa) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/SPFTest.java:177]: Using mockito maybe?

```
DNSServer mockDNSService = mock(DNSService.class);
when(mockDNSService.getByName(any())).thenAnswer(...);
return mockDNSService;
```

This significantly reduces the boiler plate thus here should be favored.

**Review by chibenwa — COMMENTED** [review]
Thanks for the nice contribution.

Sorry for the late review, I am just back from vacations.

Regards

**Review by chibenwa — APPROVED** [review]
Looks perfect to me!

Thanks for the Xmas gift ;-)

Regards,

Benoit

**Comment by ouvtam** [comment]
> Looks perfect to me!
> 
> Thanks for the Xmas gift ;-)
> 
> Regards,
> 
> Benoit

Superb, thanks for the fast feedback!

I was assuming you were on vacation when there was no feedback. I hope the vacation refilled your batteries :-)

**Comment by chibenwa** [comment]
>  I hope the vacation refilled your batteries :-)

Yes, it definitly did!

**Pull request merged**